Lib/glyphsLib/builder/builders.py

```diff
@@ -197,8 +208,8 @@ def masters(self):
             # Save processing bracket layers for when designspace() is called, as we
             # have to extract them to free-standing glyphs.
             if (
-                "[" in layer.name
-                and "]" in layer.name
+                BRACKET_LAYER_RE.match(layer.name)
+                and glyph.export
```

Comment:
I added a check so that glyphs marked as non-export that contain bracket layers do not get processed as such; the layers are exported as regular layers to the UFO, not as separate bracket glyphs

Reply:
Why? Could you add a comment to explain why?

Reply:
oh sure I can. I thought it made sense, but I can take it out if you think it's not a good idea. I was thinking, a non-export glyph may be one which contains unfinished work that is not interpolatable; and unbalanced bracket layers must be so, because they implicitly reference master layers. So marking the glyph as non-export should not prevent to build a varfont.
I guess with the new skipExportGlyphs feature this is not an issue because sources are subsetted before interpolation/compilation; but the old subsetting feature in fontmake happens after compilation.

Reply:
oh now I remember another reason. You don't want to add desingspace rules referencing glyphs that are not going to be there in the exported font.

Reply:
Cool, that makes total sense, I was just wondering why